version: '3.8'

services:

  db:
    container_name: ToDoManager_bd
    image: postgres:14.5-alpine
    volumes:
      - postgres_data:/var/lib/postgresql/data
    restart: always
    environment:
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_DB: ${DB_NAME}
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U $DB_USER" ]
      interval: 5s
      timeout: 5s
      retries: 5
    networks:
      - back_nw
  api:
    container_name: ToDoManager_api
    image: egrigorev/todomanager:${TAG_NAME:-latest}
    restart: always
    environment:
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_DB: ${DB_NAME}
      HOST: ${DB_HOST}
    volumes:
      - .env:/app/.env
    depends_on:
      db:
        condition: service_healthy
    command: python todo_manager/manage.py runserver 0.0.0.0:8000
    networks:
      - back_nw
      - front_nw
  front:
    container_name: ToDoManager_front
    image: sermalenk/skypro-front:lesson-37
    restart: always
    ports:
      - "80:80"
    volumes:
      - ./nginx/default.conf:/etc/nginx/conf.d/default.conf
      - ./static:/opt/static/
    depends_on:
      api:
        condition: service_started
    networks:
      - front_nw
volumes:
  postgres_data:
networks:
  back_nw:
    driver: bridge
  front_nw:
    driver: bridge